File: lib/features/budget/domain/models/budget_limit.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:savvy/core/constants/financial_enums.dart';

part 'budget_limit.freezed.dart';
part 'budget_limit.g.dart';

@freezed
abstract class BudgetLimit with _$BudgetLimit {
  const factory BudgetLimit({
    required String id,
    required ExpenseCategory category,
    required double monthlyLimit,
    @Default(true) bool isActive,
    required DateTime createdAt,
    @Default(false) bool isDeleted,
  }) = _BudgetLimit;

  factory BudgetLimit.fromJson(Map<String, dynamic> json) =>
      _$BudgetLimitFromJson(json);
}

File: SavvyApp/Packages/Core/SavvyFoundation/Sources/SavvyFoundation/Models/BudgetLimit.swift
import Foundation

public struct BudgetLimit: FirestoreEntity {
    public let id: String
    public var category: ExpenseCategory
    public var monthlyLimit: Decimal
    public var isActive: Bool
    public let createdAt: Date
    public var isDeleted: Bool

    public init(
        id: String = UUID().uuidString,
        category: ExpenseCategory,
        monthlyLimit: Decimal,
        isActive: Bool = true,
        createdAt: Date = Date(),
        isDeleted: Bool = false
    ) {
        self.id = id
        self.category = category
        self.monthlyLimit = monthlyLimit
        self.isActive = isActive
        self.createdAt = createdAt
        self.isDeleted = isDeleted
    }
}
